app/src/main/java/com/mobileproto/hireddit/hireddit/speech/SpeechListener.java

```diff
@@ -8,83 +8,69 @@
 import java.util.ArrayList;
 
 /**
- * Voice Recognition: For speech to text input
+ * Voice Recognition: Listener for speech to text input
  */
 
-public class SpeechListener implements RecognitionListener
-{
-    String DEBUG_TAG = "SpeechListener Debug";
-    private String voiceInput;
+public class SpeechListener implements RecognitionListener {
+    String DEBUG_TAG = "myDebug";
     private ArrayList data;
     private ArrayList partial;
     private SpeechCallback speechCallback;
-
-    public SpeechListener(SpeechCallback mSpeechCallback)
-    {
+    private int numErrors = 0;
+    public SpeechListener(SpeechCallback mSpeechCallback) {
         speechCallback = mSpeechCallback;
     }
 
 
     @Override
-    public void onReadyForSpeech(Bundle params)
-    {
+    public void onReadyForSpeech(Bundle params) {
         Log.d(DEBUG_TAG, "onReadyForSpeech " + params);
     }
 
     @Override
-    public void onBeginningOfSpeech()
-    {
+    public void onBeginningOfSpeech() {
         Log.d(DEBUG_TAG, "oneBeginningOfSpeech - Start talking!");
     }
 
     @Override
-    public void onRmsChanged(float rmsdB)
-    {
+    public void onRmsChanged(float rmsdB) {
         //Log.d(DEBUG_TAG, "onRmsChanged " + rmsdB);
     }
 
     @Override
-    public void onBufferReceived(byte[] buffer)
-    {
+    public void onBufferReceived(byte[] buffer) {
         Log.d(DEBUG_TAG, "onBuffer Received");
     }
 
     @Override
-    public void onEndOfSpeech()
-    {
+    public void onEndOfSpeech() {
         Log.d(DEBUG_TAG, "onEndOfSpeech - Finished talking.");
+        numErrors = 0;
```

Comment:
does this ever get called even when there is an error? Ie, are we extraneously setting `numErrors=0`?